gc/base/standard/Scavenger.cpp

```diff
@@ -3072,25 +3073,27 @@ MM_Scavenger::abandonSurvivorTLHRemainder(MM_EnvironmentStandard *env)
 }
 
 void
-MM_Scavenger::abandonTenureTLHRemainder(MM_EnvironmentStandard *env)
+MM_Scavenger::abandonTenureTLHRemainder(MM_EnvironmentStandard *env, bool preserveRemainders)
 {
 	if (NULL != env->_tenureTLHRemainderBase) {
 		Assert_MM_true(NULL != env->_tenureTLHRemainderTop);
-		env->_scavengerStats._tenureDiscardBytes += (uintptr_t)env->_tenureTLHRemainderTop - (uintptr_t)env->_tenureTLHRemainderBase;
 		_tenureMemorySubSpace->abandonHeapChunk(env->_tenureTLHRemainderBase, env->_tenureTLHRemainderTop);
-		env->_tenureTLHRemainderBase = NULL;
-		env->_tenureTLHRemainderTop = NULL;
+
+		if (!preserveRemainders){
+			env->_scavengerStats._tenureDiscardBytes += (uintptr_t)env->_tenureTLHRemainderTop - (uintptr_t)env->_tenureTLHRemainderBase;
+			env->_tenureTLHRemainderBase = NULL;
+			env->_tenureTLHRemainderTop = NULL;
+		}
+		/* In case of Mutator threads (for concurrent scavenger) isMasterThread() is not sufficient, we must also make a thread check with getThreadType()*/
+		else if ((env->isMasterThread()) && (GC_MASTER_THREAD == env->getThreadType())){
+			saveTenureTLHRemainders(env);
```

Comment:
for symmetry between save and restore, I'd move 

+			env->_tenureTLHRemainderBase = NULL;
+			env->_tenureTLHRemainderTop = NULL;

to saveTenureTLHRemainders()